docs/openapi.yml

```diff
@@ -0,0 +1,1367 @@
+      summary: Password Reset
+      description: Allows a user to reset their password using a unique token.
+      security: []
+      parameters:
+        - in: path
+          name: token
+          required: true
+          schema:
+            type: string
+            description: The unique token for password reset verification.
+            example: "0cc376fa-0433-42f3-a600-6e04967cf955"
+            pattern: ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                password:
+                  type: string
+                  example: Xyz@1234e3
+                  pattern: ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$
+                  description: The new password.
+                  maxLength: 255
+                repassword:
+                  type: string
+                  example: Xyz@1234e3
+                  pattern: ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$
+                  description: The new password, repeated for confirmation.
+                  maxLength: 255
+      responses:
+        '204':
+          description: Password has been reset successfully.
+        '400':
+          $ref: '#/components/responses/BadRequestError'
+        '404':
+          $ref: '#/components/responses/NotFoundError'
+        '401':
+          $ref: '#/components/responses/UnauthorizedError'
+        '406':
+          $ref: '#/components/responses/NotAcceptableError'
+        '415':
+          $ref: '#/components/responses/UnsupportedMediaType'
+        '429':
+          $ref: '#/components/responses/TooManyRequests'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+
+  /users:
+    get:
+      tags:
+        - User Management
+      summary: List users with pagination
+      description: Retrieves a paginated list of users.
+      parameters:
+        - in: query
+          name: page
+          schema:
+            type: integer
+            default: 1
+            minimum: 1
+            format: int32
+          description: The current page number.
+        - in: query
+          name: per_page
+          schema:
+            type: integer
+            maximum: 100
+            minimum: 1
+            format: int32
+            default: 15
+          description: The number of items to return per page.
+      responses:
+        '200':
+          description: A paginated list of users.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/Pagination'
+                  - type: object
+                    properties:
+                      data:
+                        type: array
+                        items:
+                          type: object
```

Comment:
Question, can we refer to a typed object? It looks like it's repeating the same field names over and over

Reply:
I don't quite understand you